File: leetcode-submissions/0083-remove-duplicates-from-sorted-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        //set
        Set<Integer> vals = new HashSet<>();
        
        if (head == null || head.next == null) {
            return head;
        }
        ListNode prev = head;
        ListNode curr = head.next;
        vals.add(prev.val);
        while(curr != null) {
            int val = curr.val;
            if (vals.contains(val)) {
                prev.next = curr.next;
                curr = curr.next;
            } else {
                vals.add(val);
                prev = curr;
                curr = curr.next;
            } 
        }
        // 1, 2, 1, 4, 2
        //set = 1, 2, 4,
        return head;  
    }
}

File: leetcode-submissions/0083-remove-duplicates-from-sorted-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:

        dummy = ListNode(val=float("-inf"))
        dummy.next = head

        prev, curr = dummy, head

        while curr is not None:
            if prev.val == curr.val:
                temp_next = curr.next
                prev.next = temp_next
                curr = temp_next
            else:
                prev = curr
                curr = curr.next

        return dummy.next

